### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by 1688aa** [Eveldee/KeePass-ExeIconPicker#1]
Hope to support Android apk import icon

**Comment by Eveldee** [comment]
I don't really know how to do that, if anyone has an idea, it will help me a lot.
I guess I can just extract the .apk file as a zip and try to get the icon resource, but maybe there's a better solution.

**Comment by Darthagnon** [comment]
There's a plugin for QuickLook that can view the icons for APKs: https://github.com/canheo136/QuickLook.Plugin.ApkViewer

**Comment by Eveldee** [comment]
> There's a plugin for QuickLook that can view the icons for APKs: https://github.com/canheo136/QuickLook.Plugin.ApkViewer

That's very informative and show that it's not impossible to do, but he uses a custom wrapper around aapt that I can't use because the code is licensed under the [GPL-3.0 License](https://github.com/canheo136/QuickLook.Plugin.ApkViewer/blob/master/LICENSE) and I want to keep my code under MIT.

I may look into making my own wrapper for aapt, all the other wrappers that I found are not updated anymore

**Comment by Darthagnon** [comment]
[APK Info](https://github.com/Enyby/APK-Info) also extracts metadata, including icons, from APKs. I'm not sure what license it uses (moved between multiple devs, no license file) but it's written in AutoIt, which I believe is fairly readable code - maybe there's some easy implementation that could be deduced from it?

EDIT: If you do add APK icon support, would it be possible to also add XAPK support (zipped multi-arch APK standard from APKPure, supported by BlueStacks, APKPure (which is "trusted" by Huawei stores and the APK Info project)? (structure is a zip file with multiple APK files, OBB files, a manifest, and a PNG icon)

EDIT 2: APK Info also uses aapt binaries, not sure what wrapper they use

**Comment by Eveldee** [comment]
There's also another solution: let the user write the app name and it downloads the app icon from the Play Store, I just don't know if they would allow scrapping their app icons like this, maybe doing it from another site like APKMirror would be better.

I'll look into using aapt when I'll have some time, that seems to be the best solution so far

**Comment by Darthagnon** [comment]
I had a thought the after posting my last comment - I was posting info without thinking "Is this really necessary?". There's also another solution, let the user manually extract the APK using 7zip and browse to the PNG using the icon selection window already present in KeePass, if they're desperate for an APK icon within KeePass. Much better than internet connections or bloating the plugin with extra stuff most people won't use. 

Basically no change to the plugin as-is lol. Maybe a text noting the manual icon extraction process in the About >> Plugin equivalent in KeePass.

'cause, to be honest, APKs are meant for a platform (Android) other than the one this plugin is intended to be used on (Windows). If it were me, I wouldn't bother implementing the feature.

**Comment by Eveldee** [comment]
That's what I thought at first but it's not that easy to extract icon from applications now, apks are indeed zip archives but icons are now hidden in compressed ressources now, that's why it needs tools like aapt. If it were simply a zip extract, I would implement it using available libraries in .Net Framework, but it's harder than that.

Anyway, I'll lock this issue till someone find a lightweight solution to extract icons from apks, aapt would take ~1 MB while the plugin is less than 20 KB, it is indeed bloating the plugin as you said.
